extensions/kubernetes-client/runtime-internal/src/main/java/io/quarkus/kubernetes/client/runtime/KubernetesClientUtils.java

```diff
@@ -1,18 +1,48 @@
 package io.quarkus.kubernetes.client.runtime;
 
 import java.time.Duration;
+import java.util.Objects;
+import java.util.ServiceLoader;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import java.util.stream.StreamSupport;
 
 import org.eclipse.microprofile.config.ConfigProvider;
 
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import io.fabric8.kubernetes.api.model.KubernetesResource;
 import io.fabric8.kubernetes.client.Config;
 import io.fabric8.kubernetes.client.ConfigBuilder;
 import io.fabric8.kubernetes.client.KubernetesClient;
 import io.fabric8.kubernetes.client.KubernetesClientBuilder;
+import io.fabric8.kubernetes.client.utils.KubernetesSerialization;
+import io.fabric8.kubernetes.internal.KubernetesDeserializer;
 import io.quarkus.runtime.TlsConfig;
 
 public class KubernetesClientUtils {
 
     private static final String PREFIX = "quarkus.kubernetes-client.";
+    private static final ObjectMapper KUBERNETES_CLIENT_MAPPER = new ObjectMapper();
+    // Unless we grab a static reference to the KubernetesResource classes, native mode will fail when deserializing
+    // since those classes won't be available and won't be registered in the KubernetesSerialization instance
+    private static final Set<Class<? extends KubernetesResource>> KUBERNETES_RESOURCE_CLASES = scanKubernetesResources();
+    private static final KubernetesSerialization SERIALIZATION = new KubernetesSerialization(KUBERNETES_CLIENT_MAPPER, false);
+    static {
```

Comment:
We should probably add all initialization in this block, otherwise the code is hard to follow

Reply:
Same as above: shouldn't this use the injectable version instead of a static version?